.github/workflows/tests.yml

```diff
@@ -2,6 +2,10 @@ name: Tests
 
 on:
   pull_request:
+  push:
+  schedule:
+    # Run all tests at midnight.
+    - cron: "0 0 * * *"
```

Comment:
Can we add some logic to run this only when we have changes in the last 24-hours? e.g.
```bash
git log --since "1 day ago" --format=oneline master | wc -l
```

Reply:
We can do quite easily, but I really think that is over-complicating things. There isn't a benefit I can see to only running it under those conditions: the CI is free, we're not lacking in capacity or likely to run into limits at midnight.

Running it regardless of the last time a commit was made gives us more timely information about issues with any of our test dependencies.

In general I think we should keep any Actions we add simple and vanilla to begin with. We can add optimisations later if needed.

Reply:
I agree that we have a commit on master almost each day but it's not the case on stable branches (which we want to move to GitHub actions in the future), e.g. on `stable/2.2.x` we can have a commit once per few months. That's why I would like to have a ready and tested solution in this trial :pray: 

> the CI is free, we're not lacking in capacity or likely to run into limits at midnight.

Yes it's free for us, but not for out planet :seedling: :world_map: If can avoid wasting energy :factory: and resource, why not.

Reply:
Ahh right. So cronjob events only trigger on the default project branch, not on other branches.

If we wanted to trigger daily tests on those branches we'd need something a bit different, but we can do that later.

The planetary impact of spinning up a VM to run "git log" isn't _much_ different from spinning up a VM to run "npm test", but I'll add the check regardless 👌

Reply:
> The planetary impact of spinning up a VM to run "git log" isn't _much_ different from spinning up a VM to run "npm test", but I'll add the check regardless ok_hand

Yeah, but it should be significant for https://djangoci.com/job/django-2.2/ :wink:

Reply:
Thanks for doing this :1st_place_medal:

Reply:
No problem! I've looked into this now - Github have specific protections against accidentally causing recursive workflow runs (see https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token).

I've pushed my implementation: We have a job that runs on a schedule and loops over a list of branches. If any of the branches have a commit in the last day it triggers a new build for that branch using the API. This requires an access token to be configured outside of Github Actions, but once done it should work. I'll test it on my fork first, and we can look at merging it later.

Reply:
With this we can also trigger branch builds. I think you need push access to the repository, but it could be useful.

![Screenshot 2021-02-11 at 19 15 17](https://user-images.githubusercontent.com/1027207/107686761-87445600-6c9d-11eb-8940-53fd6f755b5b.png)

Reply:
Done! Just tested this and it works. The access token needs to have the `Workflow` scope.

Reply:
It's not in this PR anymore. What was the reason for removing `nightly_run`? :thinking:

Reply:
Errr.... I'm not entirely sure. That might have been a mistake. I'll re-add it now 👍

Reply:
Done, see comment below